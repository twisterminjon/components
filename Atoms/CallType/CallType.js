import React from 'react'
import PropTypes from 'prop-types'

import CallIncoming from '../../Atoms/Icons/CallIncoming'
import CallOutgoing from '../../Atoms/Icons/CallOutgoing'
import CallMissed from '../../Atoms/Icons/CallMissed'
import { CallTypes, CallTypeList } from '../../../constants'

CallType.propTypes = {
  /** Type of call */
  type: PropTypes.oneOf(CallTypeList).isRequired,

  /** Can be dimmed */
  dimmed: PropTypes.bool,

  /** Can show a different display when missed */
  missed: PropTypes.bool,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}

CallType.defaultProps = {
  dimmed: false,
  missed: false,
  className: '',
  style: {},
}

export default function CallType({ type, dimmed, missed, className, style }) {
  const colors = {
    dimmed: 'var(--text_dimmed)',
    default: 'var(--text_default)',
    missed: 'var(--text_negative)',
  }

  let color = null
  color = dimmed ? colors.dimmed : colors.default
  if (missed) color = colors.missed

  // Return based on call type
  if (type === CallTypes.incoming) {
    return <CallIncoming color={color} className={className} style={style} />
  }

  if (type === CallTypes.outgoing) {
    return <CallOutgoing color={color} className={className} style={style} />
  }

  if (type === CallTypes.missed) {
    return <CallMissed color={color} className={className} style={style} />
  }

  return null
}
