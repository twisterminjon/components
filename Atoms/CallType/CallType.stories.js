import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, select, boolean } from '@storybook/addon-knobs'

import CallType from './CallType'
import Call from './CallType'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('CallType', () => {
  const typeOpts = {
    incoming: 'INCOMING',
    outgoing: 'OUTGOING',
  }
  const typeVal = select('type', typeOpts, Call.incoming)
  const dimmedVal = boolean('dimmed', false)
  const missedVal = boolean('missed', false)

  return <CallType type={typeVal} dimmed={dimmedVal} missed={missedVal} />
})
