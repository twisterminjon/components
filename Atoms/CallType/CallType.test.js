import React from 'react'
import CallType from './CallType'

describe('CallType', () => {
  it('matches snapshot for INCOMING', () => {
    const wrapper = window.shallow(<CallType type="INCOMING" dimmed={false} missed={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot for outgoing', () => {
    const wrapper = window.shallow(<CallType type="OUTGOING" dimmed={false} missed={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when dimmed', () => {
    const wrapper = window.shallow(<CallType type="OUTGOING" dimmed={true} missed={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when missed', () => {
    const wrapper = window.shallow(<CallType type="OUTGOING" dimmed={true} missed={true} />)
    expect(wrapper).toMatchSnapshot()
  })
})
