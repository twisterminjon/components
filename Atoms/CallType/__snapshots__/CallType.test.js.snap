// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallType matches snapshot for INCOMING 1`] = `
<CallIncoming
  className=""
  color="var(--text_default)"
  style={Object {}}
/>
`;

exports[`CallType matches snapshot for outgoing 1`] = `
<CallIncoming
  className=""
  color="var(--text_default)"
  style={Object {}}
/>
`;

exports[`CallType matches snapshot when dimmed 1`] = `
<CallIncoming
  className=""
  color="var(--text_dimmed)"
  style={Object {}}
/>
`;

exports[`CallType matches snapshot when missed 1`] = `
<CallIncoming
  className=""
  color="var(--text_negative)"
  style={Object {}}
/>
`;
